webapp/nuxt.config.js

```diff
@@ -277,6 +277,26 @@ export default {
           },
         ],
       })
+      const tagAttributesForTesting = ['data-test', ':data-test', 'v-bind:data-test']
+      ctx.loaders.vue.compilerOptions = {
+        modules: [
+          {
+            preTransformNode(abstractSyntaxTreeElement) {
+              if (!ctx.isDev) {
+                const { attrsMap, attrsList } = abstractSyntaxTreeElement
+                tagAttributesForTesting.forEach(attribute => {
+                  if (attrsMap[attribute]) {
+                    delete attrsMap[attribute]
+                    const index = attrsList.findIndex(attr => attr.name === attribute)
+                    attrsList.splice(index, 1)
```

Comment:
> [<img alt="alina-beck" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2019-12-10T12:19:43Z" title="Tuesday, December 10th 2019, 1:19:43 pm +01:00">Dec 10, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -272,6 +272,26 @@ export default {
           },
         },
       })
+      const tagAttributesForTesting = ['data-test', ':data-test', 'v-bind:data-test']
+      ctx.loaders.vue.compilerOptions = {
+        modules: [
+          {
+            preTransformNode(astEl) {
+              if (!ctx.isDev) {
+                const { attrsMap, attrsList } = astEl
+                tagAttributesForTesting.forEach(attribute => {
+                  if (attrsMap[attribute]) {
+                    delete attrsMap[attribute]
+                    const index = attrsList.findIndex(attr => attr.name === attribute)
+                    attrsList.splice(index, 1)
```


This could be written as:
```
attrsList = attrsList.filter(attr => !tagAttributesForTesting.includes(attr.name))
tagAttributesForTesting.forEach(attribute => delete attrsMap[attribute])
```

Instead of finding the `index` of each attribute you could just filter the list once. And `delete` just returns `true` if the attribute doesn't exist, so we don't really need the `if statement`. :wink:

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-12-10T13:13:42Z" title="Tuesday, December 10th 2019, 2:13:42 pm +01:00">Dec 10, 2019</time>_

---


thanks for approving @alina-beck 
I thought the same thing when I saw the code, filter is great for this...
I tried to get it working now and it didn't though... maybe we can merge and figure out why together. 
one difference in `filter` and `splice` is that the latter changes the array in place whereas the former you need to assign to a new array.
When I tried this :point_up_2: it threw an error in `yarn build` 
```
TypeError: Assignment to constant variable.
```
So, I attempted to make it a let, but it failed to remove the `data-test` attibutes... maybe it would allow to create a new array and then set the `attrsList` array to the new? Not sure.

I tried without the if clause as well, and it built and started, but then threw this error 
```
Cannot read property '_normalized' of undefined
```
So, in the end the only thing I have changed is to make the variable more verbose.
Thoughts?

Reply:
> [<img alt="alina-beck" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2019-12-10T14:06:39Z" title="Tuesday, December 10th 2019, 3:06:39 pm +01:00">Dec 10, 2019</time>_

---


Interesting... 🙊

You're right, it looks like we're not allowed to assign to `attrsList` which makes `splice` the better (only?) choice. The `_normalized` issue I don't understand – `delete` really shouldn't complain about anything? But well, we have a working solution and it's easy enough to read, so I'm happy to just leave it as is! :smile: